SConstruct

```diff
@@ -731,11 +731,24 @@ elif env.msvc:
         )
         Exit(255)
 
-# Default architecture flags.
-if env["arch"] == "x86_32":
-    if env.msvc:
+# Set x86 CPU instruction sets to use by the compiler's autovectorization.
+if env["arch"] == "x86_64":
+    # On 64-bit x86, enable SSE 4.2 and prior instruction sets (SSE3/SSSE3/SSE4/SSE4.1) to improve performance.
+    # This is supported on most CPUs released after 2009-2011 (Intel Nehalem, AMD Bulldozer).
+    # AVX and AVX2 aren't enabled because they aren't available on more recent low-end Intel CPUs.
+    if env.msvc and not methods.using_clang(env):
+        # https://stackoverflow.com/questions/64053597/how-do-i-enable-sse4-1-and-sse3-but-not-avx-in-msvc/69328426
+        env.Append(CCFLAGS=["/d2archSSE42"])
+    else:
+        env.Append(CCFLAGS=["-msse2", "-msse4.2"])
```

Comment:
Why pass in `-msse2` here? Isn't it part of the x64 instruction set? Even if it wasn't, wouldn't it be implied by `-msse4.2`?

Reply:
It *should* be implied, but I need to check again with elfx86exts to be 100% sure.

Reply:
As seen [here](https://en.wikipedia.org/wiki/X86-64#Microarchitecture_levels), SSE and SSE2 are both part of the `x86-64-v1` baseline.

The [GCC documentation](https://gcc.gnu.org/onlinedocs/gcc-15.1.0/gcc/x86-Options.html) for `-mfpmath` mentions something along these lines as well:

> For the x86-32 compiler, you must use `-march=cpu-type`, `-msse` or `-msse2` switches to enable SSE extensions and make this option effective. For the x86-64 compiler, these extensions are enabled by default.

Reply:
I can confirm the binaries are bit-for-bit identical once stripped when I remove `-msse2` from the x86_64 build options, so I went ahead and pushed that.